src/FloatingDockContainer.cpp

```diff
@@ -537,8 +537,9 @@ void CFloatingDockContainer::onDockAreasAddedOrRemoved()
 	if (TopLevelDockArea)
 	{
 		d->SingleDockArea = TopLevelDockArea;
-		d->setWindowTitle(
-		    d->SingleDockArea->currentDockWidget()->windowTitle());
+		CDockWidget* CurrentWidget = d->SingleDockArea->currentDockWidget();
+		d->setWindowTitle(CurrentWidget->windowTitle());
+		setWindowIcon(CurrentWidget->icon());

```

Comment:
Set the window icon only, if the CurrentWidget->icon() is valid - if the CurrentWidget has no icon, then set the Application icon

Reply:
Fixed